go.mod

```diff
@@ -53,6 +53,7 @@ require (
 	github.com/opencontainers/runtime-spec v1.1.1-0.20230922153023-c0e90434df2a
 	github.com/opencontainers/runtime-tools v0.9.1-0.20230914150019-408c51e934dc
 	github.com/opencontainers/selinux v1.11.0
+	github.com/peterbourgon/diskv/v3 v3.0.1
```

Comment:
I would prefer if we used something that is maintained. Peter does not seem to be actively maintaining this project any more, at least per some of the bugs there, that would be best closed, per:

- https://github.com/peterbourgon/diskv/issues/63
- https://github.com/peterbourgon/diskv/issues/66

This is a bit of a no-no in my book.

I don't want to chase future bugs in some new dependency if there is no need to.

We could use something like Badger, which has in-memory mode (good for testing), TTL on cached objects, and garbage collection, and even supports encryption (if we ever need such a thing). However, it's a large project primarily focusing on performance, and it's not memory-usage friendly, sadly.

There is also the bbolt, a fork by us, by CoreOS team, to keep BoltDB going. This is something that etcd is using at the moment.

Thoughts?

Reply:
I decided against bolt because it's not really made for larger files and does not represent the files on disk in a 1:1 mapping. The maintainability is tricky here, right. Maybe we should implement it on our own? Or just use the parts of diskv we actually require?